### ISSUE TITLE
No way to keyboard focus on extension management buttons

### ISSUE THREAD

**Issue by ejizba** [microsoft/vscode#120202]
Issue Type: <b>Bug</b>

Open up any extension's marketplace page in VS Code like this:

![Screen Shot 2021-03-30 at 1 45 57 PM](https://user-images.githubusercontent.com/11282622/113055165-1e3a7400-915f-11eb-8fbc-aed1f7486ecc.png)

Try to tab or arrow key to the "Disable", "Uninstall" or "Manage" (the gear icon) buttons. Instead, it skips from "License" at the top right to "Details" at the bottom left

VS Code version: Code - Insiders 1.55.0-insider (c185983a683d14c396952dd432459097bc7f757f, 2021-03-30T12:47:00.900Z)
OS version: Darwin x64 20.3.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 2, 2|
|Memory (System)|16.00GB (0.86GB free)|
|Process Argv|--crash-reporter-id 63154ab5-ba0e-419a-aa36-bd67d86db27f|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by sandy081** [comment]
@isidorn These actions are using toolbar. Any insights why they are not able to be accessed by keyboard?

**Comment by isidorn** [comment]
@sandy081 I can reproduce this issue, though this should just work.
I think the problem are those action with the dropdown. They are `ActionWithDropdownActionViewItem` and that class does not seem to handle focus properly.
It should be able to be focused and to be `setFocusable`.

**Comment by sandy081** [comment]
@isidorn Can you please shed more light on how `ActionWithDropdownActionViewItem` has to handle focus properly? It is extending `ActionViewItem` which I think is handling focus properly right?

**Comment by isidorn** [comment]
@sandy081 you are correct, I changed the actionBar in the extensions list to only focus on enabled items and to be focusable.
So now this works.
What is needed are two fixes on your side:
* When the focus is on the `ActionWithDropdownActionViewItem` properly render it 
* When the Extension has not install or rating make those elements not focusable (I can repro this when running out of source)

**Comment by alexr00** [comment]
Somehow the "disable" button gets skipped over, but only sometimes. After visiting the "disable" button once, it will then get tabbed to correctly. 
![Uploading recording (2).gif…]()

**Comment by sandy081** [comment]
@alexr00 This is as designed because of new action bar behaviour. You can navigate actions in the actions bar only with right/left arrows. If you tab/shift+tab you will be taken out of action bar. So when you enter action bar please use arrows to navigate. 

CC @isidorn

**Comment by bpasero** [comment]
I find it weird though that the first thing to focus is the disable button and not the left most button:

![image](https://user-images.githubusercontent.com/900690/116657933-857e5a80-a98f-11eb-9bc7-2cbed55b5233.png)

**Comment by isidorn** [comment]
@bpasero the first thing to focus in a toolbar should be the left most button (even if it is disabled).
Maybe the order in the extensions view is a bit strange...

**Comment by sandy081** [comment]
Yeah its weird why it is not getting focused on the left most button, will investigate.

**Comment by fsteffi** [comment]
GitHubTags:#A11yTCS;#Win32;#DesktopApp;#A11y_VSCode;#WCAG2.1.1;#A11ySev1;#CAITestPass_May2020;#Visual Studio Code;#A11yMAS;#Keyboard;
